src/server/set_family.cc

```diff
@@ -994,41 +994,45 @@ OpResult<StringVec> OpScan(const OpArgs& op_args, string_view key, uint64_t* cur
   return res;
 }
 
-void SendNumeric(OpResult<uint32_t> result, SinkReplyBuilder* builder) {
+void SendNumeric(OpResult<uint32_t> result, CommandContext* cmd_cntx) {
   switch (result.status()) {
     case OpStatus::OK:
-      return builder->SendLong(result.value());
+      return cmd_cntx->SendLong(result.value());
     case OpStatus::WRONG_TYPE:
-      return builder->SendError(kWrongTypeErr);
+      return cmd_cntx->SendError(kWrongTypeErr);
     default:
-      return builder->SendLong(0);
+      return cmd_cntx->SendLong(0);
   }
 }
 
 struct SetReplies {
-  SetReplies(CommandContext* cntx, bool _script) : cmd_cntx(cntx), script(_script) {
+  explicit SetReplies(CommandContext* cntx)
+      : cmd_cntx(cntx), script(cntx->server_conn_cntx()->conn_state.script_info) {
   }
 
-  template <typename T> void Send(vector<T>* sv) {
+  template <typename T> void Send(vector<T> sv) {
     if (script)  // output is sorted under scripts
-      sort(sv->begin(), sv->end());
-
-    static_cast<RedisReplyBuilder*>(cmd_cntx->rb())->SendBulkStrArr(*sv, CollectionType::SET);
+      sort(sv.begin(), sv.end());
+    auto replier = [vec = std::move(sv)](facade::SinkReplyBuilder* builder) {
+      auto* rb = static_cast<RedisReplyBuilder*>(builder);
+      rb->SendBulkStrArr(vec, CollectionType::SET);
+    };
+    cmd_cntx->ReplyWith(std::move(replier));
```

Comment:
btw you can make this lambda accept a RedisReplyBuilder directly

Reply:
what do you mean? change `ReplyWith` implementation or add another method like ReplyWithResp?